### PR TITLE
[python] replace misuses of `typing.NoReturn`

### LINKED ISSUE #10429: [🐛 Bug]: Python NoReturn type in Chromium Options
### What happened?

For the Python ChromeOptions, NoReturn is used  incorrectly and throws Pylance off. 

Python Selenium Version: 4.1.0 was working fine (no return types)
Python Selenium Version: 4.1.3 pylance error shows up.

For the functions where NoReturn is used incorrectly, the return type should be changed to None.

### How can we reproduce the issue?

```shell
from selenium.webdriver import ChromeOptions

class CustomOptions(ChromeOptions):
    def __init__(self, page_load_strategy = 'normal', headless=True) -> None:
        super().__init__()

        if headless:
            self.add_argument('headless')  
            self.add_argument('window-size=1920,1080')
        else:
            self.add_argument('start-maximized')

        self.add_experimental_option('useAutomationExtension', False)
        self.add_argument('--ignore-certificate-errors')
        self.add_argument('--incognito')
        self.add_argument('--disable-blink-features=AutomationControlled')
        self.add_argument('--log-level=3')
        self.page_load_strategy = page_load_strategy
```

### Relevant log output

```shell
Pylance:

(method) add_argument: (argument: Any) -> None
Adds an argument to the list

:Args:

Sets the arguments
Code is unreachable Pylance
```

### Operating System

Windows 11

### Selenium version

Python selenium v 4.1.3

### What are the browser(s) and version(s) where you see this issue?

Chrome 99

### What are the browser driver(s) and version(s) where you see this issue?

Chromedriver version 99

### Are you using Selenium Grid?

None

### PR DESCRIPTION
**Pull request by symonk**
Recent introduction of typing has a few incorrect use cases; This patch addresses misuses of `typing.NoReturn` which should be used when a function has no possibility of returning, mostly used incorrectly in place of `None` on various setters.  I will audit the other uses of `NoReturn` throughout; consider this a start for now.

closes #10429 (eventually)

### Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [X] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: py/selenium/webdriver/chromium/options.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ def extensions(self) -> List[str]:
             file_.close()
         return encoded_extensions + self._extensions
```
Comment (symonk) [inline py/selenium/webdriver/chromium/options.py:86]: type this correctly

File: py/selenium/webdriver/chromium/options.py [hunk 2]
```diff
@@ -142,7 +142,7 @@ def headless(self) -> bool:
         return '--headless' in self._arguments
 
     @headless.setter
-    def headless(self, value: bool) -> NoReturn:
```
Comment (symonk) [inline py/selenium/webdriver/chromium/options.py:145]: ` -> bool:`

File: py/selenium/webdriver/chromium/options.py [hunk 3]
```diff
@@ -59,7 +59,7 @@ def debugger_address(self: str) -> str:
         return self._debugger_address
```
Comment (symonk) [inline py/selenium/webdriver/chromium/options.py:59]: type this

File: py/selenium/webdriver/chromium/options.py [hunk 4]
```diff
@@ -122,7 +122,7 @@ def experimental_options(self) -> dict:
         """
         return self._experimental_options
```
Comment (symonk) [inline py/selenium/webdriver/chromium/options.py:123]: type this

File: py/selenium/webdriver/chromium/options.py [hunk 5]
```diff
@@ -142,7 +142,7 @@ def headless(self) -> bool:
         return '--headless' in self._arguments
```
Comment (symonk) [inline py/selenium/webdriver/chromium/options.py:142]: `-> bool`

File: py/selenium/webdriver/ie/webdriver.py [hunk 6]
```diff
@@ -115,7 +114,7 @@ def __init__(self, executable_path=DEFAULT_EXECUTABLE_PATH, capabilities=None,
             keep_alive=keep_alive)
         self._is_remote = False
 
-    def quit(self) -> NoReturn:
+    def quit(self) -> None:
         RemoteWebDriver.quit(self)
```
Comment (symonk) [inline py/selenium/webdriver/ie/webdriver.py:118]: Todo: consider `super()` in this class in general, will address in a seperate ticket

**Review by AutomatedTester — APPROVED** [review]

**Pull request merged**